core/src/main/resources/META-INF/resources/primefaces-extensions/suneditor/suneditor-widget.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * PrimeFaces SunEditor component
+ * API reference http://suneditor.com/sample/html/out/document-editor.html
+ *
+ * @author Matthieu Valente
+ */
+PrimeFaces.widget.ExtSunEditor = PrimeFaces.widget.DeferredWidget.extend({
+
+    /**
+     * Initializes the widget.
+     *
+     * @param {object} cfg The widget configuration.
+     */
+    init: function(cfg) {
+        this._super(cfg);
+        this.input = $(this.jqId);
+        this.disabled = (cfg.disabled === undefined) ? false : cfg.disabled;
+
+        if (this.disabled) {
+            this.input.attr("disabled", "disabled");
+            this.jq.addClass("ui-state-disabled");
+            this.cfg.readOnly = true;
+        }
+
+        if (this.cfg.buttonListJSON) {
+            this.cfg.buttonList = JSON.parse(this.cfg.buttonListJSON.replace(/'/g, "\""));
+        }
+
+        // user extension to configure handsontable
+        var extender = this.cfg.extender
+        if (extender) {
+            if (typeof extender === "function") {
+                extender.call(this);
+            } else {
+                PrimeFaces.error("Extender value is not a javascript function!");
+            }
+        }
+
+        this.renderDeferred();
+    },
+
+    /**
+     * Initializes the SunEditor instance. This method will be called when the
+     * resources for the SunEditor are loaded.
+     *
+     * @private
+     */
+    _render: function() {
+        var $this = this;
+
+        //initialize
+        this.getLanguage();
+        this.editor = SUNEDITOR.create(this.id, this.cfg);
+        if (this.cfg.readOnly) {
+            this.editor.readOnly(true);
+        }
+        if (this.disabled) {
+            this.editor.disable();
+        }
+
+        //update input on change
+        this.editor.onChange = function(contents, core) {
+            $this.input.val([].map.call(core.context.element.wysiwyg.children, function(node) {
+                return node.outerHTML || "";
+            }).join(""));
+            $this.callBehavior('change');
```

Comment:
The behaviors are not documented in the taglib xml.

Reply:
Ajax events should end up in the documentation right? We get them from the taglib if I'm right.

<img width="469" alt="Screenshot 2023-04-04 at 15 05 42" src="https://user-images.githubusercontent.com/7500178/229801118-d7352e74-2689-422d-9ebb-e302fd868d8b.png">

Reply:
I'll add the events to the showcase tomorrow as well as a locale.

Reply:
OK cool the locales I include in the JS under /lang we just need to figure out how to load one!  I think it might be similar to what you have done in other plugins.